addons/binding/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/discovery/RioSystemDeviceDiscoveryService.java

```diff
@@ -0,0 +1,270 @@
+import org.openhab.binding.russound.internal.rio.controller.RioControllerConfig;
+import org.openhab.binding.russound.internal.rio.source.RioSourceConfig;
+import org.openhab.binding.russound.internal.rio.system.RioSystemHandler;
+import org.openhab.binding.russound.internal.rio.zone.RioZoneConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This implementation of {@link DiscoveryService} will scan a RIO device for all controllers, source and zones attached
+ * to it.
+ *
+ * @author Tim Roberts
+ *
+ */
+public class RioSystemDeviceDiscoveryService extends AbstractDiscoveryService {
+    /** The logger */
+    private final static Logger logger = LoggerFactory.getLogger(RioSystemDeviceDiscoveryService.class);
+
+    /** The system handler to scan */
+    private final RioSystemHandler sysHandler;
+
+    /** Pattern to identify controller notifications */
+    private final Pattern RSP_CONTROLLERNOTIFICATION = Pattern.compile("(?i)^[SN] C\\[(\\d+)\\]\\.(\\w+)=\"(.*)\"$");
+
+    /** Pattern to identify source notifications */
+    private final Pattern RSP_SRCNOTIFICATION = Pattern.compile("(?i)^[SN] S\\[(\\d+)\\]\\.(\\w+)=\"(.*)\"$");
+
+    /** Pattern to identify zone notifications */
+    private final Pattern RSP_ZONENOTIFICATION = Pattern
+            .compile("(?i)^[SN] C\\[(\\d+)\\]\\.Z\\[(\\d+)\\]\\.(\\w+)=\"(.*)\"$");
+
+    /**
+     * The {@link SocketSession} that will be used to scan the device
+     */
+    private SocketSession _session;
+
+    /**
+     * The {@link WaitingSessionListener} to the {@link #_session} to receive/process responses
+     */
+    private WaitingSessionListener _listener;
+
+    /**
+     * Create the discovery service from the {@link RioSystemHandler}
+     *
+     * @param sysHandler a non-null {@link RioSystemHandler}
+     * @throws IllegalArgumentException if sysHandler is null
+     */
+    public RioSystemDeviceDiscoveryService(RioSystemHandler sysHandler) {
+        super(RussoundHandlerFactory.SUPPORTED_THING_TYPES_UIDS, 30, false);
+
+        if (sysHandler == null) {
+            throw new IllegalArgumentException("sysHandler can't be null");
+        }
+        this.sysHandler = sysHandler;
+    }
+
+    /**
+     * Activates this discovery service. Simply registers this with
+     * {@link RioSystemHandler#registerDiscoveryService(RioSystemDeviceDiscoveryService)}
+     */
+    public void activate() {
+        sysHandler.registerDiscoveryService(this);
+    }
+
+    /**
+     * Deactivates the scan - will disconnect the session and remove the {@link #_listener}
+     */
+    @Override
+    public void deactivate() {
+        if (_session != null) {
+            try {
+                _session.disconnect();
+            } catch (IOException e) {
+                // ignore
+            }
+            _session.removeListener(_listener);
+            _session = null;
+            _listener = null;
+        }
+
+        // sysHandler.unregisterDiscoveryService();
```

Comment:
I did this - however, just food for thought.  I fully disagree with this standard.  Commented code has alot of value to someone reading the code (including the original author) since it's like code that was commented out either for a specific reason or for code that could potentially be used in the future.  I see it as very much like documentation.  But again, don't really care about this one - so I did it (any any others that I coudl find)